helios-testing/src/main/java/com/spotify/helios/testing/ExistingHeliosDeployment.java

```diff
@@ -0,0 +1,237 @@
+      log.warn("HeliosClient did not close cleanly: {}", e);
+    }
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /**
+   * Undeploys and deletes jobs leftover from previous runs of TemporaryJobs. This would happen if
+   * the test was terminated before the cleanup code was called. This method will iterate over each
+   * file in the specified directory. Each filename is the prefix that was used for job names
+   * during previous runs. The method will undeploy and delete any jobs that have a matching
+   * prefix, and the delete the file. If the file is locked, it is currently in use, and will be
+   * skipped.
+   * @throws ExecutionException
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @VisibleForTesting
+  void removeOldJobs(final List<File> files)
+      throws ExecutionException, InterruptedException, IOException {
+    if (files == null || files.size() == 0) {
+      return;
+    }
+
+    log.info("Removing old temporary jobs");
+
+    final Map<JobId, Job> jobs = heliosClient.jobs().get();
+
+    // Iterate over all files in the directory
+    for (final File file : files) {
+      // Skip .tmp files which are generated when JobPrefixFiles are created. Also skip
+      // directories. We don't expect any, but skip them just in case.
+      if (file.getName().endsWith(".tmp") || file.isDirectory()) {
+        continue;
+      }
+      // If we can't obtain a lock for the file, it either has already been deleted, or is being
+      // used by another process. In either case, skip over it.
+      try (
+          final JobPrefixFile prefixFile = JobPrefixFile.tryFromExistingFile(file.toPath())
+      ) {
+        if (prefixFile == null) {
+          log.debug("Unable to create JobPrefixFile for {}", file.getPath());
+          continue;
+        }
+
+        boolean jobRemovalFailed = false;
+        // Iterate over jobs, looking for ones with a matching prefix.
+        for (final Map.Entry<JobId, Job> entry : jobs.entrySet()) {
+          final JobId jobId = entry.getKey();
+          // Skip over job if the id doesn't start with current filename.
+          if (!jobId.getName().startsWith(prefixFile.prefix())) {
+            continue;
+          }
+          // Get list of all hosts where this job is deployed, and undeploy
+          final JobStatus status = heliosClient.jobStatus(entry.getKey()).get();
+          final List<String> hosts = ImmutableList.copyOf(status.getDeployments().keySet());
+          final List<AssertionError> errors = undeployer.undeploy(entry.getValue(), hosts);
+
+          // Set flag indicating if any errors occur
+          if (!errors.isEmpty()) {
+            jobRemovalFailed = true;
+          }
+        }
+
+        // If all jobs were removed successfully, then delete the prefix file. Otherwise,
+        // leave it there so we can try again next time.
+        if (!jobRemovalFailed) {
+          prefixFile.delete();
```

Comment:
This calls into `Files.deleteIfExists()` which according to its javadocs:

```
     * @return  {@code true} if the file was deleted by this method; {@code
     *          false} if the file could not be deleted because it did not
     *          exist
 ```

Does it matter if it's false in this case?